azure/scope/managedcontrolplane.go

```diff
@@ -706,3 +710,34 @@ func (s *ManagedControlPlaneScope) AvailabilityStatusFilter(cond *clusterv1.Cond
 	}
 	return cond
 }
+
+// PrivateEndpointSpecs returns the private endpoint specs.
+func (s *ManagedControlPlaneScope) PrivateEndpointSpecs() []azure.ResourceSpecGetter {
+	privateEndpointSpecs := make([]azure.ResourceSpecGetter, 0)
```

Comment:
I suspect the intent here was to include the capacity in the `make()` call:
```suggestion
	privateEndpointSpecs := make([]azure.ResourceSpecGetter, 0, len(s.ControlPlane.Spec.VirtualNetwork.Subnet.PrivateEndpoints))
```